### ISSUE TITLE
unix domain socket support

### ISSUE THREAD

**Issue by slingamn** [kiwiirc/webircgateway#20]
I'd be excited about support for UNIX domain sockets in both directions:

1. as the listen socket for `webircgateway`
1. as the upstream ircd socket that it connects to (as of oragono/oragono#187, oragono's trunk supports listening on a UNIX domain socket)

**Comment by slingamn** [comment]
Yes; we're using nginx as a reverse proxy in front of webircgateway:

https://github.com/darwin-network/slash/blob/4410dc54f97147265e18808228a65069b5127ba9/etc/nginx/sites-available/https#L34

so we'd prefer to have nginx and webircgateway communicating via UNIX domain socket instead of loopback TCP.

**Comment by prawnsalad** [comment]
Ah OK, I was originally thinking of incoming IRC connections which confused me. HTTP listening on unix sockets now implemented in master, https://github.com/kiwiirc/webircgateway/commit/3703dc9fce2bd93745aa34c91864e66eb93b3695

**Comment by slingamn** [comment]
I tested this on irc.darwin.network and ran into a couple snags:

1. The default socket permissions are 0755; since the nginx role user (`www-data:www-data`) and our oragono rule user (`oragono:nogroup`) are different, nginx can't connect. A config option to set the permissions would be helpful? I could submit a PR for this.
1. When the incoming connection to webircgateway is on a UNIX domain socket, it emits an invalid `WEBIRC` command:

````
write(7, "WEBIRC yzuke-FyYuBLVjGlnpgbPw websocketgateway <nil> <nil> remote-port\n", 71) = 71
````

whereas when it listens on loopback TCP, the `WEBIRC` command is correct even when the upstream ircd is on a UNIX domain socket:

````
write(7, "WEBIRC yzuke-FyYuBLVjGlnpgbPw websocketgateway google-public-dns-a.google.com 8.8.8.8 :secure remote-port=37084\n", 112) = 112
````

(I substituted a well-known IP address for my home IP; password intentionally not redacted.)
